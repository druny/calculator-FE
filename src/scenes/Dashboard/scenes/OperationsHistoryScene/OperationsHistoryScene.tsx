import React, { useEffect, useState } from 'react';

import { makeStyles } from '@material-ui/core/styles';
import LinearProgress from '@material-ui/core/LinearProgress';
import TableContainer from '@material-ui/core/TableContainer';
import Table from '@material-ui/core/Table';
import Paper from '@material-ui/core/Paper';

import { TableBody } from './components/TableBody';
import { TableHeader } from './components/TableHeader';

import { getAllHistoryAdapter } from './services/apiAdapters';

type HistoryType = {
  leftNumber: string;
  operation: string;
  rightNumber: string;
  _id: string;
};

const useStyles = makeStyles({
  table: {
    marginTop: 25,
    maxHeight: '50vh',
  },
});

const cells = ['LeftNumber', 'Operation', 'RightNumber'];

export function OperationsHistoryScene() {
  const [, setIsLoading] = useState(false);
  const [history, setHistory] = useState<HistoryType[]>([]);

  const classes = useStyles();

  useEffect(() => {
    async function getHistory() {
      setIsLoading(true);

      const historyResponse: any = await getAllHistoryAdapter();

      historyResponse.reverse();
      historyResponse.length = 10;

      setHistory(historyResponse);

      setIsLoading(false);
    }

    setInterval(() => {
      getHistory();
    }, 5000);

    getHistory();
  }, []);

  return (
    <TableContainer component={Paper} className={classes.table}>
      <LinearProgress variant="determinate" value={100} color="secondary" />
      <Table stickyHeader>
        <TableHeader cells={cells} />
        <TableBody history={history} />
      </Table>
    </TableContainer>
  );
}
