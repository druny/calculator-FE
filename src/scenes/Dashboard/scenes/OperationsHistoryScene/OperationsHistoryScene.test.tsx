import React from 'react';
import renderer from 'react-test-renderer';

import { OperationsHistoryScene } from './OperationsHistoryScene';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<OperationsHistoryScene />).toJSON();
  expect(tree).toMatchSnapshot();
});
