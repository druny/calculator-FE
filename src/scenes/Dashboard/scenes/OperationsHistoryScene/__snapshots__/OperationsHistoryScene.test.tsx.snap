// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root makeStyles-table-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    aria-valuenow={100}
    className="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
      style={
        Object {
          "transform": "translateX(0%)",
        }
      }
    />
  </div>
  <table
    className="MuiTable-root MuiTable-stickyHeader"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
          scope="col"
        >
          LeftNumber
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
          scope="col"
        >
          Operation
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
          scope="col"
        >
          RightNumber
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    />
  </table>
</div>
`;
